crates/pallet-domains/src/staking_epoch.rs

```diff
@@ -0,0 +1,884 @@
+        // of new epoch and before operator de-registered
+        for (nominator_id, _) in PendingDeposits::<T>::drain_prefix(operator_id) {
+            T::Currency::thaw(&freeze_identifier, &nominator_id)
+                .map_err(|_| TransitionError::RemoveLock)?;
+        }
+
+        // remove OperatorOwner Details
+        OperatorIdOwner::<T>::remove(operator_id);
+
+        Ok(())
+    })
+    .map_err(Error::UnlockOperator)
+}
+
+fn unlock_nominator_withdrawals<T: Config>(
+    operator_id: OperatorId,
+    current_consensus_block: T::BlockNumber,
+) -> Result<(), Error> {
+    let freeze_identifier = T::FreezeIdentifier::staking_freeze_id(operator_id);
+    match PendingNominatorUnlocks::<T>::take(operator_id, current_consensus_block) {
+        None => Ok(()),
+        Some(withdrawals) => withdrawals.into_iter().try_for_each(|withdrawal| {
+            let frozen_balance =
+                T::Currency::balance_frozen(&freeze_identifier, &withdrawal.nominator_id);
+
+            let remaining_staked_balance = frozen_balance
+                .checked_sub(&withdrawal.balance)
+                .ok_or(TransitionError::BalanceUnderflow)?;
+
+            T::Currency::set_freeze(
+                &freeze_identifier,
+                &withdrawal.nominator_id,
+                remaining_staked_balance,
+            )
+            .map_err(|_| TransitionError::UpdateLock)?;
+
+            Ok(())
+        }),
+    }
+    .map_err(Error::UnlockNominator)
+}
+
+#[derive(TypeInfo, Debug, Encode, Decode, Clone, PartialEq, Eq)]
+pub struct PendingNominatorUnlock<NominatorId, Balance> {
+    pub nominator_id: NominatorId,
+    pub balance: Balance,
+}
+
+fn do_finalize_domain_pending_transfers<T: Config>(
+    domain_id: DomainId,
+    current_consensus_block: T::BlockNumber,
+) -> Result<(), Error> {
+    DomainStakingSummary::<T>::try_mutate(domain_id, |maybe_stake_summary| {
+        let stake_summary = maybe_stake_summary
+            .as_mut()
+            .ok_or(TransitionError::MissingDomainStakeSummary)?;
+
+        let next_epoch = stake_summary
+            .current_epoch_index
+            .checked_add(One::one())
+            .ok_or(TransitionError::EpochOverflow)?;
+
+        let mut total_domain_stake = BalanceOf::<T>::zero();
+        let mut current_operators = BTreeMap::new();
+        for next_operator_id in &stake_summary.next_operators {
+            let operator_stake = finalize_operator_pending_transfers::<T>(
+                *next_operator_id,
+                current_consensus_block,
+            )?;
+
+            total_domain_stake = total_domain_stake
+                .checked_add(&operator_stake)
+                .ok_or(TransitionError::BalanceOverflow)?;
+            current_operators.insert(*next_operator_id, operator_stake);
+        }
+
+        stake_summary.current_epoch_index = next_epoch;
+        stake_summary.current_total_stake = total_domain_stake;
+        stake_summary.current_operators = current_operators;
```

Comment:
Also need to set `next_operators` here

```suggestion
        stake_summary.current_operators = current_operators.clone();
        stake_summary.next_operators = current_operators;
```

Reply:
not required, the `next_operators` are always upto date. No need to update them here again

Reply:
Well, I only see `next_operators` being removed or inserted for register/deregister requests, but `next_operators` should be set to `current_operators` at the beginning of a epoch, no?

Reply:
> When current_operators is updated from next_operators

Sorry, I think I missed this.